### ISSUE TITLE
False positive on invoke where src is a machine factory function

### ISSUE THREAD

**Issue by VanTanev** [rlaffers/eslint-plugin-xstate#7]
**Describe the bug**
```ts
const createInvokeMachine = () => createMachine({ /* ... */ })

const machine = createMachine({
  invoke: {
    // error: [eslint xstate/invoke-usage] The` value of the "src" property in the "invoke" object must be a machine, function, string or an object.
    src: createInvokeMachine()
  },
});
```

**Expected behavior**
Machine creation functions should be allowed as an argument to `src`.

**Discussion**
I am not sure how powerful ESlint is here - can it actually track that a creation function returns a machine, across imports? It seems like it won't be able to.

Maybe it's fine to allow the rule to accept a function call expression?

**Versions (please complete the following information):**
 - Node version: `v14.17.3`
 - ESLint version: `7.23.0`
 - eslint-plugin-xstate version: `0.12.2`

**Comment by rlaffers** [comment]
> I am not sure how powerful ESlint is here - can it actually track that a creation function returns a machine, across imports? It seems like it won't be able to.
That's right. Since this is a static code analysis tool (it builds an AST) ESlint does not execute functions to track down the return value.
The best we can do here is to allow functions as a `src` value. The message actually says functions are allowed so I'm going to check why this fails.

**Comment by VanTanev** [comment]
Thanks for the lightning-fast reaction!

Hope you have a great day :)

**Comment by rlaffers** [comment]
No problem and thank you for the report!
